### PR TITLE
Added images to numpy convert file

### PR DESCRIPTION
**Pull request by Shakib-IO**

### CODE REVIEW

File: reusable/images_to_numpy.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
+import os
+import cv2
+import numpy as np
+from tqdm import tqdm
+from PIL import Image
+import matplotlib.pyplot as plt
+
+# Assuming Folder Structure
```
Comment (hasibzunair) [inline reusable/images_to_numpy.py]: Script should have only a single use-case of what it is going to do. 

See [this](https://github.com/hasibzunair/plugnplay/blob/main/reusable/letterbox_resize.py) for writing function description.

File: reusable/images_to_numpy.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    return img
+
+# Read images and convert it to RGB
+def get_data(images):
+    img = cv2.imread(images)
+    img = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)  # Convert BGR to RGB
+    img = crop_and_resize(img)
+    return img
+
+# Load images and convert it to numpy
```
Comment (hasibzunair) [inline reusable/images_to_numpy.py]: `Load images and convert it to numpy` should be inside the function.

File: reusable/images_to_numpy.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    inp_feat = []
+    image_files = os.listdir(path)
+    for file_name in tqdm(image_files):
+        file_path = os.path.join(path, file_name)
+        img = get_data(file_path)
+        inp_feat.append(img)
+
+    inp_feat = np.array(inp_feat)
+    print(inp_feat.shape)
+    np.save(f"{folder_name}.npy", inp_feat)  # Save the .npy file with the folder name
```
Comment (hasibzunair) [inline reusable/images_to_numpy.py]: Save folder should be taken from user as input, not hard coded.

File: reusable/images_to_numpy.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    print("Done!")
+
+def main() -> None:
+
+    data_path = "./Datasett/"
```
Comment (hasibzunair) [inline reusable/images_to_numpy.py]: data_path should be provided as input.

File: reusable/images_to_numpy.py [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+    image_path = os.path.join(data_path, "Images")
```
Comment (hasibzunair) [inline reusable/images_to_numpy.py]: also `Images`.

File: reusable/images_to_numpy.py [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+
+    convert_to_numpy(image_path)
+
+
+
+"""Optional"""
```
Comment (hasibzunair) [inline reusable/images_to_numpy.py]: take input from user whether to visualize outputs, then use the functions. Use [this](https://github.com/hasibzunair/plugnplay/blob/main/reusable/argparsers.py) for taking input from users.

File: reusable/images_to_numpy.py [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+  plt.show()
+
+
+if __name__ == "__main__":
+    main()
+    
+    """if you want to see your converted .npy file as image."""
+    image_data = np.load("Images.npy")
+    
+    if image_data.shape[0] == 1:
```
Comment (hasibzunair) [inline reusable/images_to_numpy.py]: make a new function called `visualize`, the rest of this code should go there.

**Review by hasibzunair — CHANGES_REQUESTED** [review]

File: reusable/images_to_numpy.py (hunk 1 above)
Comment (Shakib-IO) [reply reusable/images_to_numpy.py]: Done!!

File: reusable/images_to_numpy.py [hunk 8]
```diff
@@ -1,34 +1,76 @@
...
 import cv2
+import argparse
 import numpy as np
 from tqdm import tqdm
 from PIL import Image
 import matplotlib.pyplot as plt
 
-# Assuming Folder Structure
 """
+Assuming Dataset Folder Structure.
```
Comment (hasibzunair) [inline reusable/images_to_numpy.py]: Do like this: 

"""
Script to convert a folder of images into numpy format.

Folder structure for images:
Dataset:
    - Images
        - image_01
        - image_02
        - image_03
"""

File: reusable/images_to_numpy.py [hunk 9]
```diff
@@ -1,34 +1,76 @@
...
+
 Dataset:
     - Images
         - image_01
         - image_02
         - image_03
+
+Example:
```
Comment (hasibzunair) [inline reusable/images_to_numpy.py]: Usage instead of example

File: reusable/images_to_numpy.py [hunk 10]
```diff
@@ -1,34 +1,76 @@
...
+Example: 
+python images_to_numpy.py --root_data_path "./Dataset" --image_path "Images" --saved_folder_name "to_numpy" --visualize yes --visualize_for multiple 
 """
-# Resize the image
-def crop_and_resize(img, resize_dim=1024):
+
+parser = argparse.ArgumentParser(description= 'Convert images to numpy array')
+parser.add_argument('--root_data_path', type= str, required= True, default = "./Dataset", help = 'Define root dataset path.')
+parser.add_argument('--image_path', type= str, required= True, default= "Images", help = 'Set images path.')
+parser.add_argument('--saved_file_name', type= str, required= True, default= "images_to_nparray", help = 'Define saved file name.')
```
Comment (hasibzunair) [inline reusable/images_to_numpy.py:26]: code gives this error: 

```
check.py: error: the following arguments are required: --saved_file_name
```

**Review by hasibzunair — CHANGES_REQUESTED** [review]

**Comment by hasibzunair** [comment]
@Shakib-IO always test your code and make sure that it works locally before making changes in git. This problem should have been caught if you ran the usage command you added in the script.

File: reusable/images_to_numpy.py (hunk 10 above)
Comment (Shakib-IO) [reply reusable/images_to_numpy.py:26]: Fixed the issues.

File: reusable/images_to_numpy.py [hunk 11]
```diff
@@ -0,0 +1,133 @@
...
+    - image_01
+    - image_02
+    - image_03
+
+Usage: 
+python images_to_numpy.py --root_data_path "./Dataset" --image_path "Images" --saved_file_name "to_numpy" --visualize yes --visualize_for multiple 
+"""
+
+parser = argparse.ArgumentParser(description= 'Convert images to numpy array')
+parser.add_argument('--root_data_path', type= str, required= True, default = "./Dataset", help = 'Define root dataset path.')
```
Comment (hasibzunair) [inline reusable/images_to_numpy.py]: just use `image_path`, remove `root_data_path`. Users will define full path to images in `image_path`.

File: reusable/images_to_numpy.py [hunk 12]
```diff
@@ -0,0 +1,133 @@
...
+parser.add_argument('--image_path', type= str, required= True, default= "Images", help = 'Set images path.')
+parser.add_argument('--saved_file_name', type= str, required= True, default= "images_to_nparray", help = 'Define saved file name.')
+parser.add_argument('--visualize', type= str, choices= ['no', 'yes'], default= "no", required= True, help = 'Visualize images.')
```
Comment (hasibzunair) [inline reusable/images_to_numpy.py]: required=False, should be optional if users just wants to get numpy

File: reusable/images_to_numpy.py [hunk 13]
```diff
@@ -0,0 +1,133 @@
...
+        img: Orginal input image
+        resized_dim: Size of new image (Default: 256)
+    
+    Returns:
+        img: new image
+    """
+    img = cv2.resize(img, (resize_dim, resize_dim), interpolation=cv2.INTER_AREA)
+    return img
+
+def get_data(images):
```
Comment (hasibzunair) [inline reusable/images_to_numpy.py]: should be image_path and not images, we cannot read images from images hahah

**Review by hasibzunair — CHANGES_REQUESTED** [review]
Minor fixes needed. Almost done!

File: reusable/images_to_numpy.py (hunk 11 above)
Comment (Shakib-IO) [reply reusable/images_to_numpy.py]: Completed.

File: reusable/images_to_numpy.py (hunk 12 above)
Comment (Shakib-IO) [reply reusable/images_to_numpy.py]: Fixed!

File: reusable/images_to_numpy.py (hunk 13 above)
Comment (Shakib-IO) [reply reusable/images_to_numpy.py]: Done.

**Comment by hasibzunair** [comment]
Good job!

**Pull request merged**